specification/recoveryservicesbackup/RecoveryServices.Management/models.tsp

```diff
@@ -0,0 +1,8322 @@
+  jobType: string;
+}
+
+/**
+ * Base class for operation result info.
+ */
+#suppress "@azure-tools/typespec-azure-core/composition-over-inheritance" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+model OperationResultInfoBaseResource extends OperationWorkerResponse {
+  /**
+   * OperationResultInfoBaseResource operation
+   */
+  operation?: OperationResultInfoBase;
+}
+
+/**
+ * Base class for operation result info.
+ */
+@discriminator("objectType")
+model OperationResultInfoBase {
+  /**
+   * This property will be used as the discriminator for deciding the specific types in the polymorphic chain of types.
+   */
+  #suppress "@azure-tools/typespec-azure-core/no-string-discriminator" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+  objectType: string;
+}
+
+/**
+ * This is the base class for operation result responses.
+ */
+model OperationWorkerResponse {
+  /**
+   * HTTP Status Code of the operation.
+   */
+  statusCode?: HttpStatusCode;
+
+  /**
+   * HTTP headers associated with this operation.
+   */
+  #suppress "@azure-tools/typespec-azure-resource-manager/arm-no-record" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+  headers?: Record<string[]>;
+}
+
+/**
+ * List of ProtectedItem resources
+ */
+#suppress "@azure-tools/typespec-azure-core/composition-over-inheritance" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+model ProtectedItemResourceList extends ResourceList {
+  /**
+   * List of resources.
+   */
+  @pageItems
+  value?: ProtectedItemResource[];
+}
+
+/**
+ * Base class for validate operation request.
+ */
+model ValidateOperationRequestResource {
+  /**
+   * Recovery point ID.
+   */
+  id: string;
+
+  /**
+   * ValidateOperationRequestResource properties
+   */
+  properties: ValidateOperationRequest;
+}
+
+/**
+ * Base class for validate operation request.
+ */
+@discriminator("objectType")
+model ValidateOperationRequest {
+  /**
+   * This property will be used as the discriminator for deciding the specific types in the polymorphic chain of types.
+   */
+  #suppress "@azure-tools/typespec-azure-core/no-string-discriminator" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+  objectType: string;
+}
+
+#suppress "@azure-tools/typespec-azure-core/documentation-required" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+model ValidateOperationsResponse {
+  /**
+   * Base class for validate operation response.
+   */
+  validateOperationResponse?: ValidateOperationResponse;
+}
+
+/**
+ * Base class for validate operation response.
+ */
+model ValidateOperationResponse {
+  /**
+   * Gets the validation result
+   */
+  @OpenAPI.extension("x-ms-identifiers", #["code"])
+  validationResults?: ErrorDetail[];
+}
+
+/**
+ * Error Detail class which encapsulates Code, Message and Recommendations.
+ */
+model ErrorDetail {
```

Comment:
duplicate with common types
<img width="301" height="29" alt="image" src="https://github.com/user-attachments/assets/609639d9-da25-47b6-9fa6-fa2bd562136d" />

Reply:
The service used this type in their models. JS needs to rename it with client customization.